Exception-wrapping in Coder breaks SynchronousChainedCombinedDriver under Flink

While trying to do a fairly simple outer join job:
```scala
val left = readCSV[LeftRecord](leftInputPath)
val right = readCSV[RightRecord](leftInputPath)

val leftByKey = left.map(v => v.key -> v)
val rightByKey = right.flatMap(v => v.maybeKey.map(key => key -> v))

// key is of type "case class Isrc(value: String)" which has appropriate Kryo

val joined = leftByKey.fullOuterJoin(rightByKey)

joined
   .map { case (key, pair) => pair.toString }
   .saveAsTextFile("foo.txt")
sc.close()
```
on very small inputs (40-70K lines each, 0 to 1 lines per key, very rarely a couple more), the job crashes quickly under Flink (both local and remote)

The immediately apparent error is an `EOFException`, even though the fact is not immediately obvious upon inspection:

```
Caused by: org.apache.flink.runtime.client.JobExecutionException: Job execution failed.
	at org.apache.flink.runtime.jobmaster.JobResult.toJobExecutionResult(JobResult.java:146)
	at org.apache.flink.client.program.rest.RestClusterClient.submitJob(RestClusterClient.java:265)
	... 22 common frames omitted
Caused by: com.spotify.scio.coders.CoderException: null
	at org.apache.flink.runtime.io.disk.SimpleCollectingOutputView.nextSegment(SimpleCollectingOutputView.java:79)
```  

> (yes, the report says it's a `CoderException` yet the code at https://github.com/apache/flink/blob/master/flink-runtime/src/main/java/org/apache/flink/runtime/io/disk/SimpleCollectingOutputView.java#L79 clearly says `EOFException` )

```text
	at org.apache.flink.runtime.memory.AbstractPagedOutputView.advance(AbstractPagedOutputView.java:137)
	at org.apache.flink.runtime.memory.AbstractPagedOutputView.write(AbstractPagedOutputView.java:198)
	at org.apache.beam.runners.flink.translation.wrappers.DataOutputViewWrapper.write(DataOutputViewWrapper.java:48)
	at com.esotericsoftware.kryo.io.Output.flush(Output.java:163)
	at com.esotericsoftware.kryo.io.OutputChunked.flush(OutputChunked.java:45)
	at com.twitter.chill.TraversableSerializer.$anonfun$write$1(Traversable.scala:31)
	at com.twitter.chill.TraversableSerializer.$anonfun$write$1$adapted(Traversable.scala:27)
	at scala.collection.Iterator.foreach(Iterator.scala:937)
	at scala.collection.Iterator.foreach$(Iterator.scala:937)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1425)
	at scala.collection.IterableLike.foreach(IterableLike.scala:70)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:69)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at com.twitter.chill.TraversableSerializer.write(Traversable.scala:27)
	at com.twitter.chill.TraversableSerializer.write(Traversable.scala:21)
	at com.esotericsoftware.kryo.Kryo.writeObject(Kryo.java:523)
	at com.esotericsoftware.kryo.serializers.ObjectField.write(ObjectField.java:61)
	at com.esotericsoftware.kryo.serializers.FieldSerializer.write(FieldSerializer.java:495)
	at com.esotericsoftware.kryo.Kryo.writeClassAndObject(Kryo.java:599)
	at com.spotify.scio.coders.KryoAtomicCoder$$anonfun$encode$1.apply(KryoAtomicCoder.scala:141)
	at com.spotify.scio.coders.KryoAtomicCoder$$anonfun$encode$1.apply(KryoAtomicCoder.scala:131)
	at com.spotify.scio.coders.KryoAtomicCoder$.withKryoState(KryoAtomicCoder.scala:274)
	at com.spotify.scio.coders.KryoAtomicCoder.encode(KryoAtomicCoder.scala:131)
	at org.apache.beam.sdk.coders.Coder.encode(Coder.java:136)
	at com.spotify.scio.coders.WrappedBCoder$$anonfun$encode$2.apply$mcV$sp(Coder.scala:211)
	at com.spotify.scio.coders.WrappedBCoder$$anonfun$encode$2.apply(Coder.scala:211)
	at com.spotify.scio.coders.WrappedBCoder$$anonfun$encode$2.apply(Coder.scala:211)
	at com.spotify.scio.coders.WrappedBCoder.catching(Coder.scala:195)
	at com.spotify.scio.coders.WrappedBCoder.encode(Coder.scala:211)
	at org.apache.beam.sdk.transforms.join.UnionCoder.encode(UnionCoder.java:71)
	at org.apache.beam.sdk.transforms.join.UnionCoder.encode(UnionCoder.java:32)
	at org.apache.beam.sdk.coders.KvCoder.encode(KvCoder.java:71)
	at org.apache.beam.sdk.coders.KvCoder.encode(KvCoder.java:36)
	at org.apache.beam.sdk.util.WindowedValue$FullWindowedValueCoder.encode(WindowedValue.java:529)
	at org.apache.beam.sdk.util.WindowedValue$FullWindowedValueCoder.encode(WindowedValue.java:520)
	at org.apache.beam.sdk.util.WindowedValue$FullWindowedValueCoder.encode(WindowedValue.java:480)
	at org.apache.beam.runners.flink.translation.types.CoderTypeSerializer.serialize(CoderTypeSerializer.java:85)
	at org.apache.flink.api.java.typeutils.runtime.TupleSerializer.serialize(TupleSerializer.java:125)
	at org.apache.flink.api.java.typeutils.runtime.TupleSerializer.serialize(TupleSerializer.java:30)
	at org.apache.flink.runtime.operators.sort.NormalizedKeySorter.write(NormalizedKeySorter.java:281)
	at org.apache.flink.runtime.operators.chaining.SynchronousChainedCombineDriver.collect(SynchronousChainedCombineDriver.java:169)
	at org.apache.flink.runtime.operators.util.metrics.CountingCollector.collect(CountingCollector.java:35)
	at org.apache.flink.runtime.operators.chaining.ChainedMapDriver.collect(ChainedMapDriver.java:79)
	at org.apache.flink.runtime.operators.util.metrics.CountingCollector.collect(CountingCollector.java:35)
	at org.apache.flink.api.java.operators.translation.PlanFilterOperator$FlatMapFilter.flatMap(PlanFilterOperator.java:58)
	at org.apache.flink.runtime.operators.FlatMapDriver.run(FlatMapDriver.java:109)
	at org.apache.flink.runtime.operators.BatchTask.run(BatchTask.java:503)
	at org.apache.flink.runtime.operators.BatchTask.invoke(BatchTask.java:368)
	at org.apache.flink.runtime.taskmanager.Task.run(Task.java:704)
	... 1 common frames omitted
Caused by: com.spotify.scio.coders.CoderException$$anon$1:  - Coder was materialized at
	at com.spotify.scio.coders.CoderMaterializer$.beam(CoderMaterializer.scala:51)
	at com.spotify.scio.coders.CoderMaterializer$.beam(CoderMaterializer.scala:27)
	at com.spotify.scio.coders.CoderMaterializer$.kvCoder(CoderMaterializer.scala:80)
	at com.spotify.scio.values.PairSCollectionFunctions.toKV(PairSCollectionFunctions.scala:93)
	at com.spotify.scio.util.ArtisanJoin$.cogroupImpl(ArtisanJoin.scala:74)
	at com.spotify.scio.util.ArtisanJoin$.joinImpl(ArtisanJoin.scala:104)
	at com.spotify.scio.util.ArtisanJoin$.outer(ArtisanJoin.scala:154)
	at com.spotify.scio.values.PairSCollectionFunctions.fullOuterJoin(PairSCollectionFunctions.scala:256)
	at com.transparencyrights.Xxxxx.run(Xxxxx.scala:54)
```

Drilling down: 

1. `flink.SimpleCollectingOutputView.nextSegment()` throws `EOFException()` due to the depletion of a fixed-size memory buffer. (I have suggested https://github.com/apache/flink/pull/8981 to provide a clearer message than 'null')
2. the error is caught and rethrown a couple times as some internal Kryo buffers are rolled back, to cancel the effect of a potentially partially written Kryo object tree. During that, it is encapsulated into a KryoException which is then decapsulated as the exception leaves the Kryo stack.
3. Scio's `WrappedBCoder.catching` catches the `EOFException` (which is still the original exception thrown from `flink.SimpleCollectingOutputView.nextSegment()`, **and wraps it into a `CoderException`**
4. as we can see in the stack above, Scio's `WrappedBCoder` was actually called by BEAM's FullWindowedValueCoder, which was called by Flink's TupleSerializer
5. Flink's TupleSerializer was called by Flink's `NormalizedKeySorter.write`
6. Flink's `NormalizedKeySorter.write` is instructed to catch `EOFException`, and turns it into a "false" result. This "false" result is used by Flink's `SynchronousChainedCombineDriver.collect` to flush out some buffers, free up some memory, and carry on.

Now, due to step (3), step (6) fails and the `CoderException(EOFException) keeps traveling up the call stack, until the job is aborted.

Step (1) is a bit weird from the point of view of step (3) as the latter clearly doesn't expect a Coder to do more than flatten objects into bytes, but Flink's intricate memory management tries very hard to avoid copies and memory pressure.

I see few easy solutions here. 

**[A]** Can I modify `WrappedBCoder` so that it wraps all exceptions *except* EOFException, with a comment to the effect we sacrifice the materializer logging due to the use of EOFException for internal signalling within the Flink runner? This feels… A bit gross, but the materialization location is too useful to leave out.

**[B]** As an alternative, since `Throwable.setStackTrace()` is public, would the following make sense:
   - actually *alter and rethrow* the caught exception in `WrappedBCoder.catching`
   - the materialization stack trace gets appended at the end of the caught exception, before it gets rethrown
   - CoderException isn't private; it will have to be @deprecated for a while. There won't be any CoderException thrown anymore.

**[C]** this one is ugly, but `CoderException$` already started playing this game. Implement **[A]** or **[B]** if there is a trace of Flink anywhere in the caught call stack, otherwise keep wrapping exceptions as usual.

I can't say I really like [C]. It's a form of per-backend subclassing hidden in complexity. [A] isn't very regular, as it promotes `EOFException` to something special that escapes the normal `CoderException` wrapping, which is not useful in a non-Flink context and hopes that Flink doesn't use any other exception. [B] would be the most regular, but it means an effective change of API for all consumers. On the other hand, there is no instance of something catching specifically `scio.CoderException` in scio's code base. WDYT?

--- a/scio-core/src/main/scala/com/spotify/scio/coders/Coder.scala
+++ b/scio-core/src/main/scala/com/spotify/scio/coders/Coder.scala
@@ -17,7 +17,7 @@
 
 package com.spotify.scio.coders
 
-import java.io.{EOFException, InputStream, OutputStream}
+import java.io.{InputStream, OutputStream}
 
 import com.spotify.scio.coders.instances.Implicits
 import org.apache.beam.sdk.coders.Coder.NonDeterministicException
@@ -170,29 +170,6 @@ private final case class DisjunctionCoder[T, Id](
     }
 }
 
-final case class CoderException private[coders] (
-  stacktrace: Array[StackTraceElement],
-  cause: Throwable,
-  message: String = ""
-) extends RuntimeException {
-  override def getMessage: String = cause.getMessage
-  override def getCause: Throwable = new RuntimeException {
-    override def getMessage: String = s"$message - Coder was materialized at"
-    override def getStackTrace: Array[StackTraceElement] = stacktrace
-    override def getCause: Throwable = cause.getCause
-  }
-  override def getStackTrace: Array[StackTraceElement] = cause.getStackTrace
-}
-
-private[coders] object CoderException {
-  def prepareStackTrace: Array[StackTraceElement] =
-    Thread
-      .currentThread()
-      .getStackTrace()
-      .dropWhile(!_.getClassName.contains(CoderMaterializer.getClass.getName))
-      .take(10)
-}
-
 private[scio] final class RefCoder[T](val typeName: String, c: => BCoder[T]) extends BCoder[T] {
   def value: BCoder[T] = c
 
@@ -223,28 +200,8 @@ private[scio] case class WrappedBCoder[T](u: BCoder[T]) extends BCoder[T] {
    * Eagerly compute a stack trace on materialization
    * to provide a helpful stacktrace if an exception happens
    */
-  private val stackTrace: Array[StackTraceElement] =
-    CoderException.prepareStackTrace
-
-  private def buildException(cause: Throwable): Exception =
-    CoderException(stackTrace, cause)
-
-  private val stackSeparator =
-    new StackTraceElement(
-      "——✂——✂——✂—— Materialization point call stack follows ——✂——✂——✂——",
-      "——✂——✂——✂——",
-      "——✂——✂——✂——",
-      0
-    )
-
-  private val separatedStackTrace = Array(stackSeparator) ++ stackTrace
-
-  private def appendMaterializationStack[T <: Throwable](cause: T): T = {
-    val existingStack = cause.getStackTrace
-    val adjustedStack = existingStack ++ separatedStackTrace
-    cause.setStackTrace(adjustedStack)
-    cause
-  }
+  private val materializationStackTrace: Array[StackTraceElement] =
+    WrappedBCoder.prepareStackTrace
 
   override def toString: String = u.toString
 
@@ -252,11 +209,12 @@ private[scio] case class WrappedBCoder[T](u: BCoder[T]) extends BCoder[T] {
     try {
       a
     } catch {
-      case ex: EOFException =>
-        /* this is used under Flink for internal signalling, and caught above here. */
-        throw appendMaterializationStack(ex)
       case ex: Throwable =>
-        throw buildException(ex)
+        /* prior to scio 0.8, a wrapped exception was thrown. It is no longer the case, as some
+        backends (e.g. Flink) use exceptions as a way to signal from the Coder to the layers above
+         here; we therefore must alter the type of exceptions passing through this block.
+         */
+        throw WrappedBCoder.appendMaterializationStack(ex, None, materializationStackTrace)
     }
 
   override def encode(value: T, os: OutputStream): Unit =
@@ -284,12 +242,42 @@ private[scio] case class WrappedBCoder[T](u: BCoder[T]) extends BCoder[T] {
     u.registerByteSizeObserver(value, observer)
 }
 
-private object WrappedBCoder {
+private[scio] object WrappedBCoder {
   def create[T](u: BCoder[T]): BCoder[T] =
     u match {
       case WrappedBCoder(_) => u
       case _                => new WrappedBCoder(u)
     }
+
+  def appendMaterializationStack[X <: Throwable](
+    cause: X,
+    additionalMessage: Option[String],
+    baseStack: Array[StackTraceElement]
+  ): X = {
+    val existingStack = cause.getStackTrace
+    val messageItem =
+      additionalMessage.map(msg => new StackTraceElement("Due to " + msg, "", "", 0)).toArray
+
+    val adjustedStack = messageItem ++ existingStack ++ baseStack
+    cause.setStackTrace(adjustedStack)
+    cause
+  }
+
+  def prepareStackTrace: Array[StackTraceElement] =
+    Array(
+      new StackTraceElement(
+        "——✂——✂——✂—— Materialization point call stack follows ——✂——✂——✂——",
+        "——✂——✂——✂——",
+        "——✂——✂——✂——",
+        0
+      )
+    ) ++
+      Thread
+        .currentThread()
+        .getStackTrace()
+        .dropWhile(!_.getClassName.contains(CoderMaterializer.getClass.getName))
+        .take(10)
+
 }
 
 // Coder used internally specifically for Magnolia derived coders.

--- a/scio-core/src/main/scala/com/spotify/scio/coders/Coder.scala
+++ b/scio-core/src/main/scala/com/spotify/scio/coders/Coder.scala
@@ -17,7 +17,7 @@
 
 package com.spotify.scio.coders
 
-import java.io.{InputStream, OutputStream}
+import java.io.{EOFException, InputStream, OutputStream}
 
 import com.spotify.scio.coders.instances.Implicits
 import org.apache.beam.sdk.coders.Coder.NonDeterministicException
@@ -229,12 +229,32 @@ private[scio] case class WrappedBCoder[T](u: BCoder[T]) extends BCoder[T] {
   private def buildException(cause: Throwable): Exception =
     CoderException(stackTrace, cause)
 
+  private val stackSeparator =
+    new StackTraceElement(
+      "——✂——✂——✂—— Materialization point call stack follows ——✂——✂——✂——",
+      "——✂——✂——✂——",
+      "——✂——✂——✂——",
+      0
+    )
+
+  private val separatedStackTrace = Array(stackSeparator) ++ stackTrace
+
+  private def appendMaterializationStack[T <: Throwable](cause: T): T = {
+    val existingStack = cause.getStackTrace
+    val adjustedStack = existingStack ++ separatedStackTrace
+    cause.setStackTrace(adjustedStack)
+    cause
+  }
+
   override def toString: String = u.toString
 
   @inline private def catching[A](a: => A) =
     try {
       a
     } catch {
+      case ex: EOFException =>
+        /* this is used under Flink for internal signalling, and caught above here. */
+        throw appendMaterializationStack(ex)
       case ex: Throwable =>
         throw buildException(ex)
     }

--- a/scio-core/src/main/scala/com/spotify/scio/coders/Coder.scala
+++ b/scio-core/src/main/scala/com/spotify/scio/coders/Coder.scala
@@ -200,8 +200,8 @@ private[scio] case class WrappedBCoder[T](u: BCoder[T]) extends BCoder[T] {
    * Eagerly compute a stack trace on materialization
    * to provide a helpful stacktrace if an exception happens
    */
-  private val materializationStackTrace: Array[StackTraceElement] =
-    WrappedBCoder.prepareStackTrace
+  private[this] val materializationStackTrace: Array[StackTraceElement] =
+    CoderStackTrace.prepare
 
   override def toString: String = u.toString
 
@@ -210,11 +210,10 @@ private[scio] case class WrappedBCoder[T](u: BCoder[T]) extends BCoder[T] {
       a
     } catch {
       case ex: Throwable =>
-        /* prior to scio 0.8, a wrapped exception was thrown. It is no longer the case, as some
-        backends (e.g. Flink) use exceptions as a way to signal from the Coder to the layers above
-         here; we therefore must alter the type of exceptions passing through this block.
-         */
-        throw WrappedBCoder.appendMaterializationStack(ex, None, materializationStackTrace)
+        // prior to scio 0.8, a wrapped exception was thrown. It is no longer the case, as some
+        // backends (e.g. Flink) use exceptions as a way to signal from the Coder to the layers
+        // above here; we therefore must alter the type of exceptions passing through this block.
+        throw CoderStackTrace.append(ex, None, materializationStackTrace)
     }
 
   override def encode(value: T, os: OutputStream): Unit =
@@ -248,36 +247,6 @@ private[scio] object WrappedBCoder {
       case WrappedBCoder(_) => u
       case _                => new WrappedBCoder(u)
     }
-
-  def appendMaterializationStack[X <: Throwable](
-    cause: X,
-    additionalMessage: Option[String],
-    baseStack: Array[StackTraceElement]
-  ): X = {
-    val existingStack = cause.getStackTrace
-    val messageItem =
-      additionalMessage.map(msg => new StackTraceElement("Due to " + msg, "", "", 0)).toArray
-
-    val adjustedStack = messageItem ++ existingStack ++ baseStack
-    cause.setStackTrace(adjustedStack)
-    cause
-  }
-
-  def prepareStackTrace: Array[StackTraceElement] =
-    Array(
-      new StackTraceElement(
-        "——✂——✂——✂—— Materialization point call stack follows ——✂——✂——✂——",
-        "——✂——✂——✂——",
-        "——✂——✂——✂——",
-        0
-      )
-    ) ++
-      Thread
-        .currentThread()
-        .getStackTrace()
-        .dropWhile(!_.getClassName.contains(CoderMaterializer.getClass.getName))
-        .take(10)
-
 }
 
 // Coder used internally specifically for Magnolia derived coders.
@@ -501,4 +470,36 @@ sealed trait CoderGrammar {
 
 object Coder extends CoderGrammar with Implicits {
   @inline final def apply[T](implicit c: Coder[T]): Coder[T] = c
+}
+
+private[coders] object CoderStackTrace {
+  val CoderStackElemMarker = new StackTraceElement(
+    "### Coder materialization stack ###",
+    "",
+    "",
+    0
+  )
+
+  def prepare: Array[StackTraceElement] =
+    CoderStackElemMarker +: Thread
+      .currentThread()
+      .getStackTrace
+      .dropWhile(!_.getClassName.contains(CoderMaterializer.getClass.getName))
+      .take(10)
+
+  def append[T <: Throwable](
+    cause: T,
+    additionalMessage: Option[String],
+    baseStack: Array[StackTraceElement]
+  ): T = {
+    cause.printStackTrace()
+    val messageItem = additionalMessage.map { msg =>
+      new StackTraceElement(s"Due to $msg", "", "", 0)
+    }
+
+    val adjustedStack = messageItem ++ cause.getStackTrace ++ baseStack
+    cause.setStackTrace(adjustedStack.toArray)
+    cause
+  }
+
 }

--- a/scio-core/src/main/scala/com/spotify/scio/coders/DerivedCoder.scala
+++ b/scio-core/src/main/scala/com/spotify/scio/coders/DerivedCoder.scala
@@ -25,7 +25,11 @@ private object Derived extends Serializable {
       v
     } catch {
       case e: Exception =>
-        throw new CoderException(stack, e, msg)
+        /* prior to scio 0.8, a wrapped exception was thrown. It is no longer the case, as some
+        backends (e.g. Flink) use exceptions as a way to signal from the Coder to the layers above
+         here; we therefore must alter the type of exceptions passing through this block.
+         */
+        throw WrappedBCoder.appendMaterializationStack(e, Some(msg), stack)
     }
 
   def combineCoder[T](
@@ -41,14 +45,14 @@ private object Derived extends Serializable {
       i = i + 1
     }
 
-    val stack = CoderException.prepareStackTrace
+    val materializationStack = WrappedBCoder.prepareStackTrace
 
     @inline def destruct(v: T): Array[Any] = {
       val arr = new Array[Any](ps.length)
       var i = 0
       while (i < ps.length) {
         val p = ps(i)
-        catching(s"Error while dereferencing parameter ${p.label} in $v", stack) {
+        catching(s"Error while dereferencing parameter ${p.label} in $v", materializationStack) {
           arr.update(i, p.dereference(v))
           i = i + 1
         }
@@ -58,7 +62,9 @@ private object Derived extends Serializable {
 
     val constructor: Seq[Any] => T =
       ps =>
-        catching(s"Error while constructing object from parameters $ps", stack)(rawConstruct(ps))
+        catching(s"Error while constructing object from parameters $ps", materializationStack)(
+          rawConstruct(ps)
+        )
 
     Coder.record[T](typeName.full, cs, constructor, destruct)
   }

--- a/scio-core/src/main/scala/com/spotify/scio/coders/DerivedCoder.scala
+++ b/scio-core/src/main/scala/com/spotify/scio/coders/DerivedCoder.scala
@@ -29,7 +29,7 @@ private object Derived extends Serializable {
         backends (e.g. Flink) use exceptions as a way to signal from the Coder to the layers above
          here; we therefore must alter the type of exceptions passing through this block.
          */
-        throw WrappedBCoder.appendMaterializationStack(e, Some(msg), stack)
+        throw CoderStackTrace.append(e, Some(msg), stack)
     }
 
   def combineCoder[T](
@@ -45,7 +45,7 @@ private object Derived extends Serializable {
       i = i + 1
     }
 
-    val materializationStack = WrappedBCoder.prepareStackTrace
+    val materializationStack = CoderStackTrace.prepare
 
     @inline def destruct(v: T): Array[Any] = {
       val arr = new Array[Any](ps.length)

--- a/scio-test/src/test/scala/com/spotify/scio/coders/coders.scala
+++ b/scio-test/src/test/scala/com/spotify/scio/coders/coders.scala
@@ -413,7 +413,7 @@ class CodersTest extends FlatSpec with Matchers {
 
     val noopObserver =
       new org.apache.beam.sdk.util.common.ElementByteSizeObserver {
-        def reportElementSize(s: Long) = ()
+        def reportElementSize(s: Long): Unit = ()
       }
 
     nullBCoder.registerByteSizeObserver(nullExample1, noopObserver)
@@ -433,14 +433,12 @@ class CodersTest extends FlatSpec with Matchers {
     val ok: (String, String) = ("foo", "bar")
     val nok: (String, String) = (null, "bar")
     ok coderShould roundtrip()
-    val caught =
-      intercept[RuntimeException] {
-        nok coderShould roundtrip()
-      }
+    val caught = intercept[RuntimeException] {
+      nok coderShould roundtrip()
+    }
 
-    caught.getStackTrace.find(_.getClassName.contains(classOf[CodersTest].getName)) shouldNot be(
-      None
-    )
+    assert(caught.getStackTrace.contains(CoderStackTrace.CoderStackElemMarker))
+    assert(caught.getStackTrace.exists(_.getClassName.contains(classOf[CodersTest].getName)))
   }
 
   it should "#1651: remove all anotations from derived coders" in {

